### PR TITLE
Refine session related code: rename it, add tests, fix bugs, update docs

### PR DESCRIPTION
**Pull request by kevinbader**
## Description

- Got rid of `RigAuth` and moved the `Session` module as well as the authorization-check modules under the `RIG` namespace.
- Fix: A connection is now associated to its session right after the connection is established, given the request carries a JWT in its authorization header.
- Added unit tests for the session module.

## What to look out for

Dear reviewer, I want you to

- gain knowledge - I think it's important for you to know about the change.
- help to find bugs.

Have fun!

### CODE REVIEW

File: lib/rig/distributed_set.ex [hunk 1]
```diff
@@ -258,10 +258,11 @@ defmodule Rig.DistributedSet do
 
     n_deleted = :ets.select_delete(ets_table, match_spec)
 
-    if n_deleted > 0 do
-      Logger.debug(fn -> "Removed #{n_deleted} expired records" end)
-      RigMetrics.ControlInstrumenter.delete_blacklisted_session(n_deleted)
```
Comment (mmacai) [inline lib/rig/distributed_set.ex]: Isn't this used in Prometheus?

**Comment by mmacai** [comment]
Running smoke tests and getting logs below, can you confirm?

```
╓─────[ Running tests, plaintext Kafka connection ]─────╖

Starting zookeeper ... done
Starting rest-api  ... done
Starting kafka     ... done
08:09:37.074 module=Rig.Config [error] Could not resolve HTTPS_CERTFILE: "cert/selfsigned.pem"
08:09:37.092 module=Rig.Config [error] Could not resolve HTTPS_KEYFILE: "cert/selfsigned_key.des3.pem"
```

File: lib/rig/distributed_set.ex (hunk 1 above)
Comment (kevinbader) [reply lib/rig/distributed_set.ex]: not sure... it didn't work that way. Looking at the [instrumenter module](https://github.com/Accenture/reactive-interaction-gateway/blob/master/lib/rig_metrics/control_instrumenter.ex), it seemed like this was still work-in-progress, so I've gathered it's better to remove it and add the feature later for real. @jakobwgnr what do you think?

Comment (kevinbader) [reply lib/rig/distributed_set.ex]: should be addressed in #157

**Pull request merged**